app/models/setting/rh_cloud.rb

```diff
@@ -4,8 +4,11 @@ class Setting::RhCloud < Setting
   def self.default_settings
     return unless ActiveRecord::Base.connection.table_exists?('settings')
     return unless super
+    # If the user had auto_upload default setting, we will not surprise him, and force the value to false
+    # for new users, the default will be set to true and the value will remain nil
+    Setting.where(name: 'allow_auto_inventory_upload', default: "--- false\n...", value: nil).update_all(value: "--- false\n...")
```

Comment:
I think this would be better off in [`load_defaults`]( https://github.com/theforeman/foreman/blob/0112426cd294fda6fe65fd5f0fdb28d95024221b/app/models/setting.rb#L283).
`default_settings` shouldn't have side effects I believe.
Apart of that it looks reasonable :+1:  (read doesn't look great, but it's probably our only option xD)